data/mods/letsgo/random-teams.ts

```diff
@@ -117,88 +124,80 @@ export class RandomLetsGoTeams extends RandomTeams {
 		}
 
 		const movePool = (species.randomBattleMoves || Object.keys(this.dex.data.Learnsets[species.id]!.learnset!)).slice();
-		const moves: string[] = [];
-		const hasType: {[k: string]: true} = {};
-		hasType[species.types[0]] = true;
-		if (species.types[1]) {
-			hasType[species.types[1]] = true;
-		}
+		const types = new Set<string>();
+		types.add(species.types[0]);
+		if (species.types[1]) types.add(species.types[1]);
```

Comment:
```suggestion
		const types = new Set(species.abilities);
```

Reply:
???

Reply:
I assume this is a typo.

Reply:
The other changes look good, though, thanks!

Reply:
Yeah should be Object.values(species.abilities) my bad

Reply:
It should be `species.types`, actually; this is assigned to `types`.

Reply:
I am blind